### PR TITLE
Restrict nltk < 3.6.6 and add latest dependency checker

### PR DESCRIPTION
**Pull request by gsheni**

### CODE REVIEW

File: .github/workflows/latest_dependency_checker.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    - name: Update latest core dependencies
+      run: |
+        python -m virtualenv venv_core
+        source venv_core/bin/activate
+        python -m pip install --upgrade pip
+        python -m pip install -r requirements.txt
+        python -m pip install -r test-requirements.txt
+        make checkdeps OUTPUT_FILEPATH=nlp_primitives/tests/requirement_files/latest_core_dependencies.txt
+        cat nlp_primitives/tests/requirement_files/latest_core_dependencies.txt
+    - name: Update latest core dependencies
```
Comment (thehomebrewnerd) [inline .github/workflows/latest_dependency_checker.yml:27]: I think this should be update latests complete dependencies

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @gsheni ! I only reviewed the nltk pin so it might be good to have someone else verify the ci changes.

File: .github/workflows/latest_dependency_checker.yml (hunk 1 above)
Comment (gsheni) [reply .github/workflows/latest_dependency_checker.yml]: Done

**Review by thehomebrewnerd — APPROVED** [review]
LGTM

**Pull request merged**